pandas/core/arrays/datetimes.py

```diff
@@ -324,6 +430,26 @@ def _generate_range(cls, start, end, periods, freq, tz=None,
 
         return cls._simple_new(index.asi8, freq=freq, tz=tz)
 
+    # -----------------------------------------------------------------
+    # DatetimeLike Interface
+
+    def _unbox_scalar(self, value):
+        if not isinstance(value, (self._scalar_type, type(NaT))):
+            raise ValueError("'value' should be a a Timestamp..")
+        return value.value
+
+    def _scalar_from_string(self, value):
+        return Timestamp(value, tz=self.tz)
+
+    def _check_compatible_with(self, other):
+        if not timezones.tz_compare(self.tz, other.tz):
+            raise IncompatibleTimeZoneError(
```

Comment:
is this a new error?

Reply:
Yes, I think it's no longer needed though... An older version needed to catch just that but not anymore. I'll revert it to just a ValueError or TypeError.

Reply:
ok